explorer/helpers.go

```diff
@@ -0,0 +1,167 @@
+package explorer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+func (a *App) getNodeTwinID(nodeID string) (uint32, error) {
+	// cache node twin id for 10 mins and purge after 15
+	c := cache.New(10*time.Minute, 15*time.Minute)
```

Comment:
every time u call this method a new cache is created. so basically what happes u check "always" an empty cache then set the value on this cache, then discard it then create a new one on next call.

Instead this cache should be a state on the App instead.